crates/sui-framework/packages/sui-framework/sources/kiosk/kiosk.move

```diff
@@ -493,10 +572,15 @@ module sui::kiosk {
 
     /// Get mutable access to `profits` - useful for extendability.
     public fun profits_mut(self: &mut Kiosk, cap: &KioskOwnerCap): &mut Balance<SUI> {
-        assert!(object::id(self) == cap.for, ENotOwner);
+        assert!(has_access(self, cap), ENotOwner);
         &mut self.profits
     }
 
+    /// Get the `ID` of the Kiosk this cap is for.
+    public fun kiosk_cap_for(cap: &KioskOwnerCap): ID {
```

Comment:
This is a duplicate of `kiosk_owner_cap_for`, below.

Reply:
Nice catch! Almost missed this one.